robottelo/decorators/__init__.py

```diff
@@ -368,22 +406,29 @@ def bz_bug_is_open(bug_id):
     :rtype: bool
 
     """
-    bug = None
     try:
         bug = _get_bugzilla_bug(bug_id)
     except BugFetchError as err:
         LOGGER.warning(err)
         return False
-    # NOT_FOUND, ON_QA, VERIFIED, RELEASE_PENDING, CLOSED
-    if bug is None or bug.status not in BZ_OPEN_STATUSES:
+    else:
+        # NEW, ASSIGNED, MODIFIED, POST
+        if bug.status in BZ_OPEN_STATUSES:
+            return True
+
         # do not test bugs with whiteboard 'verified in upstream' in downstream
         # until they are in 'CLOSED' state
-        if (not settings.upstream and bug.status != 'CLOSED' and bug.whiteboard
-                and 'verified in upstream' in bug.whiteboard.lower()):
-            return True
-        return False
-    # NEW, ASSIGNED, MODIFIED, POST
-    return True
+
+        # verify all conditions are True, stopping evaluation when
+        # first condition is False
+        def skip_conditions():
+            yield not settings.upstream
+            yield _analyze_flags(bug.flags)
```

Comment:
Flags say skip, then we skip regardless of whiteboards? < I'm starting to disagree with myself ;-)